connexion/jsonref.py

```diff
@@ -0,0 +1,48 @@
+import collections
+from copy import deepcopy
+
+from jsonschema import RefResolver
+from jsonschema.exceptions import RefResolutionError  # noqa
+
+from .utils import deep_get
+
+
+def resolve_refs(spec, defns=None, store=None):
+    """
+    Resolve JSON references like {"$ref": <some URI>} in a spec.
+    Optionally takes a defns dictionary to look up references.
+    If no defns dictionary is specified, references will be resolved within
+    the spec, or resolve remotely from the web. Passing defns is useful
+    if you have a namespace collision between your spec and definitions.
+    Optionally takes a store, which is a mapping from reference URLs to a
+    dereferenced objects. Prepopulating the store can avoid network calls.
+    """
+    spec = deepcopy(spec)
+    store = store or {}
+    if defns:
+        defns = deepcopy(defns)
+    resolver = RefResolver('', spec, store)
+
+    def _do_resolve(node):
+        if isinstance(node, collections.Mapping) and '$ref' in node:
+            path = node['$ref'][2:].split("/")
+            try:
+                # resolve known references
+                store = defns if defns else spec
+                node.update(deep_get(store, path))
```

Comment:
This method `deep_get` needs to be able to resolve JSON references. In it's current form it can resolve basic paths, but there's some specific scenarios we need to take in consideration. We can simply use the `RefResolver` from jsonschema lib, no?

```python
node.update(resolver.resolve_from_url(path))
```

I took a look at the source code and seems like it can resolve local refs too: https://github.com/Julian/jsonschema/blob/master/jsonschema/validators.py#L669-L704

Reply:
I think it would need to receive the full path starting from `#/`

Reply:
note that on KeyError it will fall back to the jsonschema reference resolver.
I can remove the deep_get functionality, I'm realizing now that I kept it so that you could resolve things in case of a namespace conflict, but this only happens when resolving operations (not entire specs).

Reply:
I appreciate the review @rafaelcaricio . thanks!

Reply:
hmm.. actually removing the deep_get resolution didn't work because jsonschema RefResolver will get break on recursive definitions. I think the old behavior is actually ideal (try deep_get resolution, fall back to RefResolver).

Reply:
Hm.. ok, I haven't considered this scenario. Thank you for taking the time to work on this. I really appreciate. 👍